runtime/lua/vim/lsp/completion.lua

```diff
@@ -523,6 +705,14 @@ local function trigger(bufnr, clients, ctx)
     end
     local start_col = (server_start_boundary or word_boundary) + 1
     Context.cursor = { cursor_row, start_col }
+    if #matches > 0 and vim.o.completeopt:find('popup') then
```

Comment:
could use vim.regex to check word boundary, that's a bit less fragile incase another `popupfoo` flag is added in the future.